Source/WebCore/platform/graphics/MediaPlayer.cpp

```diff
@@ -1362,14 +1362,19 @@ void MediaPlayer::setPrivateBrowsingMode(bool privateBrowsingMode)
 // Client callbacks.
 void MediaPlayer::networkStateChanged()
 {
+    if (m_private->networkState() >= MediaPlayer::NetworkState::FormatError)
+        m_lastErrorMessage = m_private->errorMessage();
     // If more than one media engine is installed and this one failed before finding metadata,
     // let the next engine try.
     if (m_private->networkState() >= MediaPlayer::NetworkState::FormatError && m_private->readyState() < MediaPlayer::ReadyState::HaveMetadata) {
-        m_lastErrorMessage = m_private->errorMessage();
         client().mediaPlayerEngineFailedToLoad();
         if (!m_activeEngineIdentifier
             && installedMediaEngines().size() > 1
-            && (nextBestMediaEngine(m_currentMediaEngine) || nextMediaEngine(m_currentMediaEngine))) {
+            && (nextBestMediaEngine(m_currentMediaEngine) || nextMediaEngine(m_currentMediaEngine))
+#if ENABLE(MEDIA_SOURCE)
+            && !m_mediaSource
+#endif
```

Comment:
This seems like the wrong way to fix this - won't it cause any engine registered after MSE to never be used?

Reply:
As I understand it, if a MediaSource is attached to a media element, it means that only a MSE-aware player private (media engine) should be used. A non-MSE player private wouldn't know how to handle that source of data. So, if media engine trial fails for a MSE player private, why keep trying?

Are there multiple engines supporting MSE on Mac? We only have one in ports using GStreamer.

Reply:
I'm just removing this piece of code and leaving the test.